server/src/test/java/com/google/finapp/FinAppIT.java

```diff
@@ -427,4 +429,49 @@ public void createTransactionForAccount_tooLargeAmount_throwsException() throws
         IllegalArgumentException.class,
         () -> spannerDao.createTransactionForAccount(accountId, amount, isCredit));
   }
+
+  @Test
+  public void getRecentTransactionsForAccount_valid() throws Exception {
+    ByteArray fromAccountId = UuidConverter.getBytesFromUuid(UUID.randomUUID());
+    ByteArray toAccountId = UuidConverter.getBytesFromUuid(UUID.randomUUID());
+    BigDecimal fromAccountBalance = new BigDecimal(20);
+    BigDecimal toAccountBalance = new BigDecimal(0);
+    BigDecimal amount = new BigDecimal(10);
+    databaseClient.write(
+        ImmutableList.of(
+            Mutation.newInsertBuilder("Account")
+                .set("AccountId")
+                .to(fromAccountId)
+                .set("AccountType")
+                .to(AccountType.UNSPECIFIED_ACCOUNT_TYPE.getNumber())
+                .set("AccountStatus")
+                .to(AccountStatus.UNSPECIFIED_ACCOUNT_STATUS.getNumber())
+                .set("Balance")
+                .to(fromAccountBalance)
+                .set("CreationTimestamp")
+                .to(Value.COMMIT_TIMESTAMP)
+                .build(),
+            Mutation.newInsertBuilder("Account")
+                .set("AccountId")
+                .to(toAccountId)
+                .set("AccountType")
+                .to(AccountType.UNSPECIFIED_ACCOUNT_TYPE.getNumber())
+                .set("AccountStatus")
+                .to(AccountStatus.UNSPECIFIED_ACCOUNT_STATUS.getNumber())
+                .set("Balance")
+                .to(toAccountBalance)
+                .set("CreationTimestamp")
+                .to(Value.COMMIT_TIMESTAMP)
+                .build()));
+    ImmutableMap result = spannerDao.moveAccountBalance(fromAccountId, toAccountId, amount);
```

Comment:
Using `SpannerDao.moveAccountBalance` will mean that this test will also be testing `moveAccountBalance`. Our original goal for this test suite was for these tests to be as close to unit as possible, only testing one `SpannerDao` method per test (hence why we manually insert accounts). However, for this function, it'll be quite the hassle to manually do the `moveAccountBalance`, and these are technically integration tests. So it might be okay to use  `SpannerDao.moveAccountBalance`, even though it's a bit contrary to our original goals.

Thoughts on manually doing the move versus using the `SpannerDao` function?

Reply:
Ah the testing philosophy discussions :)

IMO, this current structure is fine. The ultimate goal for testing is: If code is buggy, is there a test which fails and does the test directly point to the issue. So if moveAccountBalance is buggy, since we have separate tests for moveAccountBalance, we should be able to fairly quickly pinpoint that  moveAccountBalance is failing from the dedicated tests, even though this would fail as well.

On the goal itself : "Our original goal for this test suite was for these tests to be as close to unit as possible, only testing one SpannerDao method per test". In principle, our goal should be behaviors and not methods. Most often these two coincide, but if they don't it's fine.